app/src/main/java/be/ugent/zeus/hydra/wpi/WpiActivity.java

```diff
@@ -109,6 +109,11 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         ViewModelProvider provider = new ViewModelProvider(this);
         viewModel = provider.get(WpiViewModel.class);
+        if (viewModel.getUserData().getValue() == null || !viewModel.getUserData().getValue().hasData()) {
```

Comment:
This should move to the `SuccessObserver`, starting on line 119; since we want to enable when the data updates (e.g. this check is too early, so it will never enable); you can also remove the if check then.